### PR TITLE
anime-downloader: init at 5.0.14

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes
https://github.com/anime-dl/anime-downloader

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [x] i686-linux
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WeebSorceress** [comment]
This was largely based on the flake pieces of @hlissner (https://github.com/hlissner/dotfiles/blob/master/packages/anime-downloader.nix) and @danielphan2003 (https://github.com/danielphan2003/flk/blob/main/pkgs/applications/video/anime-downloader/default.nix), so if they wish I would gladly commit their name as maintainers too.

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 176331` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anime-downloader</li>
  </ul>
</details>

**Comment by WeebSorceress** [comment]
Fixes WeebSorceress/nixpkgs#3

**Comment by teto** [comment]
` results/anime-downloader/bin/anime  gui` doesn't start but taht wouldnt be blocking.

I've first tried
```
results/anime-downloader/bin/anime watch -n
anime: anime-downloader 5.0.9
Enter a anime name or url: spy family
```
which blocked
then I've tried the example from the tutorial https://anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new
but for me it looks stuck again
```
results/anime-downloader/bin/anime watch 'code geass' -n
anime: anime-downloader 5.0.9
```
I wonder if it's due to packaging or a bug in the program ?

**Comment by danielphan2003** [comment]
> ` results/anime-downloader/bin/anime gui` doesn't start but taht wouldnt be blocking.
> 
> I've first tried
> 
> ```
> results/anime-downloader/bin/anime watch -n
> anime: anime-downloader 5.0.9
> Enter a anime name or url: spy family
> ```
> 
> which blocked then I've tried the example from the tutorial [anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new](https://anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new) but for me it looks stuck again
> 
> ```
> results/anime-downloader/bin/anime watch 'code geass' -n
> anime: anime-downloader 5.0.9
> ```
> 
> I wonder if it's due to packaging or a bug in the program ?

anime-downloader defaults to t-e-n-s-h-i.m-o-e, which is blocked by some ISPs. Switch to another provider instead by editing [`~/.config/anime-downloader/config.json`](https://anime-downlader.readthedocs.io/en/latest/usage/config.html#config-json).

anime-downloader latest release is [5.0.14](https://github.com/anime-dl/anime-downloader/releases/tag/5.0.14), so it may have some fixes for `anime gui`. I'm using the latest git version and have no problem launching the gui.

**Comment by danielphan2003** [comment]
> This was largely based on the flake pieces of @hlissner ([https://github.com/hlissner/dotfiles/blob/master/packages/anime-downloader.nix](https://github.com/hlissner/dotfiles/blob/master/packages/anime-downloader.nix?rgh-link-date=2022-06-05T03%3A20%3A31Z)) and @danielphan2003 ([https://github.com/danielphan2003/flk/blob/main/pkgs/applications/video/anime-downloader/default.nix](https://github.com/danielphan2003/flk/blob/main/pkgs/applications/video/anime-downloader/default.nix?rgh-link-date=2022-06-05T03%3A20%3A31Z)), so if they wish I would gladly commit their name as maintainers too.

I will open another PR to add myself as a maintainer in the future. For now it wouldn't be blocking.

File: pkgs/applications/video/anime-downloader/default.nix
Comment (teto) [inline pkgs/applications/video/anime-downloader/default.nix:4]: I am not a python maintainer but I think they would prefer to see these as their own packages, to avoid other packages to repackage those.

Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:4]: Thanks for the reply! So it would be best if we split these in 3 PRs (anime-downloader; pySmartDL; cfscrape)?

Comment (teto) [reply pkgs/applications/video/anime-downloader/default.nix:4]: that's it, and ideally you put yourself maintainer of those 3 even if in practice people help eachother even when they are not maintainers

Comment (teto) [reply pkgs/applications/video/anime-downloader/default.nix:4]: also with a mention as to why the tests are disabled. I am of the opinion not to overload maintainers when they add the package, and that the test can be added later but not every maintainer agrees and having a small comment can help.

**Comment by WeebSorceress** [comment]
> > ` results/anime-downloader/bin/anime gui` doesn't start but taht wouldnt be blocking.
> > I've first tried
> > ```
> > results/anime-downloader/bin/anime watch -n
> > anime: anime-downloader 5.0.9
> > Enter a anime name or url: spy family
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which blocked then I've tried the example from the tutorial [anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new](https://anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new) but for me it looks stuck again
> > ```
> > results/anime-downloader/bin/anime watch 'code geass' -n
> > anime: anime-downloader 5.0.9
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I wonder if it's due to packaging or a bug in the program ?
> 
> anime-downloader defaults to t-e-n-s-h-i.m-o-e, which is blocked by some ISPs. Switch to another provider instead by editing [`~/.config/anime-downloader/config.json`](https://anime-downlader.readthedocs.io/en/latest/usage/config.html#config-json).
> 
> anime-downloader latest release is [5.0.14](https://github.com/anime-dl/anime-downloader/releases/tag/5.0.14), so it may have some fixes for `anime gui`. I'm using the latest git version and have no problem using it.

Should I switch this package **src** from `fetchPypi` to `fetchFromGitHub` so it can get the latest release?

File: pkgs/applications/video/anime-downloader/default.nix
Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:4]: > ` results/anime-downloader/bin/anime gui` doesn't start but taht wouldnt be blocking.
> 
> I've first tried
> 
> ```
> results/anime-downloader/bin/anime watch -n
> anime: anime-downloader 5.0.9
> Enter a anime name or url: spy family
> ```
> 
> which blocked then I've tried the example from the tutorial https://anime-downlader.readthedocs.io/en/latest/usage/watch.html#anime-watch-new but for me it looks stuck again
> 
> ```
> results/anime-downloader/bin/anime watch 'code geass' -n
> anime: anime-downloader 5.0.9
> ```
> 
> I wonder if it's due to packaging or a bug in the program ?

This is my config file as an example: https://pastebin.com/raw/A5dqcsK1 (I had trouble with the default provider too)

**Comment by danielphan2003** [comment]
> Should I switch this package **src** from `fetchPypi` to `fetchFromGitHub` so it can get the latest release?

Sure, https://github.com/NixOS/nixpkgs/blob/30c3cc499d4715b050de8f4a39dbf4b8edde7e5c/doc/languages-frameworks/python.section.md#updating-packages-in-bulk allows this.

File: pkgs/applications/video/anime-downloader/default.nix
Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:4]: PRs: #176421 and #176416.

**Comment by WeebSorceress** [comment]
Won't build properly until the PR's #176416 and #176421 get merged.

File: pkgs/applications/video/anime-downloader/default.nix
Comment (teto) [inline pkgs/applications/video/anime-downloader/default.nix:32]: could you mention why ?

File: pkgs/applications/video/anime-downloader/default.nix
Comment (teto) [inline pkgs/applications/video/anime-downloader/default.nix:38]: same remark as for other pacakges.

**Comment by teto** [comment]
is this still a draft ? you may want to cleanup the history (or I could squash when merging)

File: pkgs/applications/video/anime-downloader/default.nix
Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:32]: I must fix some things. :disappointed:  

By doing checks i got this:

```sh
error: builder for '/nix/store/28cw3finqanlb9gjl55gnl3vvyqnsjd0-anime-downloader-5.0.14.drv' failed with exit code 1;
       last 10 log lines:
       >     mkdir(name, mode)
       > PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
       >
       >
       > ----------------------------------------------------------------------
       > Ran 1 test in 0.000s
       >
       > FAILED (errors=1)
       > Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
       For full logs, run 'nix log /nix/store/28cw3finqanlb9gjl55gnl3vvyqnsjd0-anime-downloader-5.0.14.drv'.
```

I could get around it by adding this line:

```nix
{
  preCheck = ''
    export HOME=$PWD/home
  '';
}
```

But then I got this error (which is probably why `anime gui` wasn't working):

```sh
error: builder for '/nix/store/kmc61n5gyhjg8vc3xwvw2wbkwg6s3fsh-anime-downloader-5.0.14.drv' failed with exit code 134;
       last 10 log lines:
       > writing requirements to anime_downloader.egg-info/requires.txt
       > writing top-level names to anime_downloader.egg-info/top_level.txt
       > reading manifest file 'anime_downloader.egg-info/SOURCES.txt'
       > adding license file 'LICENSE'
       > writing manifest file 'anime_downloader.egg-info/SOURCES.txt'
       > running build_ext
       > qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
       > This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
       >
       > /nix/store/fsskcsd0pbnlsw8j63w1ninbdcg82lb5-setuptools-check-hook/nix-support/setup-hook: line 4:   388 Aborted                 (core dumped) /nix/store/553d7c4xcwp9j1a1gb9cb1s9ry3x1pi9-python3-3.9.13/bin/python3.9 nix_run_setup test
       For full logs, run 'nix log /nix/store/kmc61n5gyhjg8vc3xwvw2wbkwg6s3fsh-anime-downloader-5.0.14.drv'.
```

So for now I need a way to "wrap" QT paths inside the package I guess...

**Comment by WeebSorceress** [comment]
> you may want to cleanup the history (or I could squash when merging)

I might have messed things up with my last commit... I'm terrible with git, sorry...

File: pkgs/applications/video/anime-downloader/default.nix
Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:32]: Seems like a common problem: https://github.com/NixOS/nixpkgs/blob/634141959076a8ab69ca2cca0f266852256d79ee/pkgs/applications/misc/openlp/lib.nix#L20-L23

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 176331` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anime-downloader</li>
  </ul>
</details>

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 176331` run on i686-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anime-downloader</li>
  </ul>
</details>

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 176331` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>anime-downloader</li>
  </ul>
</details>

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:2]:
```suggestion

```

Comment (AndersonTorres) [reply pkgs/applications/video/anime-downloader/default.nix:2]: Be explicit on the Python extensions you are using.

It is bad practice to bring whole things explicitly into the expression.

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:3]:
```suggestion
python3.pkgs.buildPythonApplication rec {
```

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:11]:
```suggestion

  src = fetchFromGitHub {
    owner = "anime-dl";
    repo = "anime-downloader";
    rev = "a509e4a8054665180bb2f172b3c0883a5a7da0ce";
    sha256 = "sha256-Uk2mtsSrb8fCD9JCFzvLBzMEB7ViVDrKPSOKy9ALJ6o=";
  };
```

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:9]: Why are you using a Git hash here, when you should use the `version` defined above?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/anime-downloader/default.nix
Comment (teto) [reply pkgs/applications/video/anime-downloader/default.nix:32]: this seems non trivial to fix so you could just reference the error to justify the disabling. Sometimes it's also possible to skip the failing tests (see other python packages on how to do this, it depends on the testing framework).

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:11]:
```suggestion
python3.pkgs.buildPythonApplication rec {
  pname = "anime-downloader";
  version = "5.0.14";

  src = fetchFromGitHub {
    owner = "anime-dl";
    repo = "anime-downloader";
    rev = version;
    sha256 = "sha256-Uk2mtsSrb8fCD9JCFzvLBzMEB7ViVDrKPSOKy9ALJ6o=";
  };
```

Comment (AndersonTorres) [reply pkgs/applications/video/anime-downloader/default.nix:11]: Lacks whitespace; and useless `(...)`

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:30]: An alternative, and more readable way to write this:

```suggestion
  propagatedBuildInputs = [
    aria2
    mpv
    nodejs
  ] ++ (with python3.pkgs; [
    beautifulsoup4
    cfscrape
    click
    coloredlogs
    fuzzywuzzy
    jsbeautifier
    pySmartDL
    pycryptodome
    pyqt5
    requests
    requests-cache
    selenium
    tabulate
  ]);
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/anime-downloader/default.nix
Comment (WeebSorceress) [reply pkgs/applications/video/anime-downloader/default.nix:31]: I was wondering how to do it like that. Thanks for the review!

File: pkgs/applications/video/anime-downloader/default.nix
Comment (AndersonTorres) [inline pkgs/applications/video/anime-downloader/default.nix:14]: Have you tried this with `buildInputs` instead of `propagatedBuildInputs`?

`propagated...` is used in some very restricted and specialized scenarios.

Indeed I would expect that only the Python packages you listed would be in `propagated` while the other 3 would be in `buildInputs`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Now squash all commits into only one.

File: maintainers/maintainer-list.nix
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:3773]: What the hell is happening here? 70 files??

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix
Comment (WeebSorceress) [reply maintainers/maintainer-list.nix:3773]: I did `git rebase --interactive 9a995f5c6a0b7e782017b725eb8a34ff0d7de527` squashing all but the last commit. Not sure what went wrong... :disappointed: 

image

**Review by AndersonTorres — APPROVED** [review]
Remember me 2 days.
If no one objects, I will merge it.

**Comment by WeebSorceress** [comment]
> Remember me 2 days. If no one objects, I will merge it.

I'll. :smile: Also, thanks for the reviews!

P.S. I went ahead and added `update.sh` so it's easier to maintain this package later.

**Comment by WeebSorceress** [comment]
@AndersonTorres, just reminding you! 😄

**Comment by AndersonTorres** [comment]
All hail the otakus!
Now I need to re-read Berserk. Four days to the new chapter!

**Pull request merged**
